### PR TITLE
ENH: Remove ivar memberless/static-only class `PrinSelf`

### PR DESCRIPTION
**Pull request by jhlegarreta**
Remove `PrinSelf` re-implementations in classes that:
- do not contain instance variable members: their parent class
  re-implementation prints the necessary information.
- contain only static variable members.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: Modules/Core/Common/include/itkShapedNeighborhoodIterator.hxx [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
- *=========================================================================*/
-#ifndef itkShapedNeighborhoodIterator_hxx
-#define itkShapedNeighborhoodIterator_hxx
-namespace itk
-{
-template <typename TImage, typename TBoundaryCondition>
-void
-ShapedNeighborhoodIterator<TImage, TBoundaryCondition>::PrintSelf(std::ostream & os, Indent indent) const
-{
-  os << indent << "ShapedNeighborhoodIterator = " << this << std::endl;
```
Comment (dzenanz) [inline Modules/Core/Common/include/itkShapedNeighborhoodIterator.hxx:26]: What will be printed now as the class name? `ConstShapedNeighborhoodIterator` or `ShapedNeighborhoodIterator`?

File: Modules/Numerics/Statistics/include/itkWeightedCovarianceSampleFilter.hxx [hunk 2]
```diff
@@ -30,17 +30,6 @@ WeightedCovarianceSampleFilter<TSample>::WeightedCovarianceSampleFilter()
   this->ProcessObject::SetNthInput(1, nullptr);
 }
 
-template <typename TSample>
-void
-WeightedCovarianceSampleFilter<TSample>::PrintSelf(std::ostream & os, Indent indent) const
-{
-  Superclass::PrintSelf(os, indent);
-  // m_Weights
-  os << indent << "Weights: " << this->GetWeightsInput() << std::endl;
```
Comment (dzenanz) [inline Modules/Numerics/Statistics/include/itkWeightedCovarianceSampleFilter.hxx:39]: Does the parent print this?

File: Modules/Numerics/Statistics/include/itkWeightedMeanSampleFilter.hxx [hunk 3]
```diff
@@ -33,17 +33,6 @@ WeightedMeanSampleFilter<TSample>::WeightedMeanSampleFilter()
   this->ProcessObject::SetNthInput(1, nullptr);
 }
 
-template <typename TSample>
-void
-WeightedMeanSampleFilter<TSample>::PrintSelf(std::ostream & os, Indent indent) const
-{
-  Superclass::PrintSelf(os, indent);
-  // m_Weights
-  os << indent << "Weights: " << this->GetWeightsInput() << std::endl;
```
Comment (dzenanz) [inline Modules/Numerics/Statistics/include/itkWeightedMeanSampleFilter.hxx:42]: Parent prints?

Comment (jhlegarreta) [reply Modules/Numerics/Statistics/include/itkWeightedMeanSampleFilter.hxx:42]: Nothing: `MeanSampleFilter` has no ivar members; the latters parent is ProcessObject. I tried to look at the definition of the method and I did not find it. Despite being really awkward, have not investigated further.

File: Modules/Numerics/Statistics/include/itkWeightedCovarianceSampleFilter.hxx (hunk 2 above)
Comment (jhlegarreta) [reply Modules/Numerics/Statistics/include/itkWeightedCovarianceSampleFilter.hxx:39]: Same comment as in https://github.com/InsightSoftwareConsortium/ITK/pull/3862/files/74c520d4231066d796ce6d83c5ba4e3d4134c6c2#r1068787233.

File: Modules/Core/Common/include/itkShapedNeighborhoodIterator.hxx (hunk 1 above)
Comment (jhlegarreta) [reply Modules/Core/Common/include/itkShapedNeighborhoodIterator.hxx:26]: Ah, haven't thought about it;  my bad. I was looking at other things when I realized about these classes having no ivar members, and remembered about this PR:
https://github.com/InsightSoftwareConsortium/ITK/pull/2715

However, it looks like all the underlying toolchain seem to make the trick for the above class:
https://github.com/InsightSoftwareConsortium/ITK/blob/23da3dfa79152648f3d38050ee147df47f41ce0a/Modules/Core/Common/test/itkBSplineInterpolationWeightFunctionTest.cxx
https://open.cdash.org/test/921810891 (see the end of the first line; the test prints other stuff without breaking the line)

vs. a class that re-implements`PrintSelf` e.g.
https://open.cdash.org/test/921810890

So it looks like the information gets printed (have not investigated the underlying mechanisms/have not checked the result of the test).

File: Modules/Numerics/Statistics/include/itkWeightedMeanSampleFilter.hxx (hunk 3 above)
Comment (dzenanz) [reply Modules/Numerics/Statistics/include/itkWeightedMeanSampleFilter.hxx:42]: `itkSetGetDecoratedInputMacro(Weights, WeightArrayType);` creates a special input to the filter. If ProcessObject prints a pointer to it, that should be fine, and we can remove this PrintSelf.

**Review by dzenanz — APPROVED** [review]

File: Modules/Numerics/Statistics/include/itkWeightedMeanSampleFilter.hxx (hunk 3 above)
Comment (jhlegarreta) [reply Modules/Numerics/Statistics/include/itkWeightedMeanSampleFilter.hxx:42]: OK, thanks Dženan. Then I understand that we're OK. Can you resolve the thread if you think we're OK or after checking it if you wish?

**Review by dzenanz — APPROVED** [review]

**Pull request merged**